src/core/account/custom-tokens.ts

```diff
@@ -32,6 +33,21 @@ const asMaybeContractLocation = asMaybe(
   })
 )
 
+/**
+ * Query the network for a token's details
+ */
+export async function getTokenDetails(
+  ai: ApiInput,
+  pluginId: string,
+  filter: EdgeGetTokenDetails
+): Promise<EdgeToken[]> {
+  // The normal code path:
+  const tools = await getCurrencyTools(ai, pluginId)
+  if (tools.getTokenDetails == null) return []
+
+  return await tools.getTokenDetails(filter)
+}
+
```

Comment:
This is simple enough, I would just put it inline in plugin-api.ts. There is no real logic in this function, just a wrapper around a tools method.